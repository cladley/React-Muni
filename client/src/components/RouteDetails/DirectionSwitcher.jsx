import React from "react";

export default function DirectionSwitcher({title, direction, onDirectionClick}) {
  return (
    <div className="direction-switcher">
      <h1>{title}</h1>
      <ul>
        {direction.map(dir => {
          return (
            <li key={dir.tag} onClick={() => onDirectionClick(dir.tag)}>
              {dir.title}
            </li>
          );
        })}
      </ul>
    </div>
  );
}
