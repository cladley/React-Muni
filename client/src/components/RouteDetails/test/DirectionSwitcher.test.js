import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import DirectionSwitcher from '../DirectionSwitcher';

describe('<DirectionSwitcher />', () => {
  let wrapper;
  const props = {
    title: '1 California',
    direction: [
      {tag: 'tag_1', title: 'Inbound to Drumm + Clay'},
      {tag: 'tag_2', title: 'Outbound to Geary + 33rd Avenue'},
    ]
  };

  beforeEach(() => {
    wrapper = shallow(<DirectionSwitcher {...props} />);
  });

  it('should be defined', () => {
    expect(wrapper).toBeDefined();
  });

  it('should render correctly', () => {
    const tree = renderer.create(<DirectionSwitcher {...props} />);
    expect(tree).toMatchSnapshot();
  });

  describe('user clicks a direction', () => {
    let onDirectionClick = jest.fn();

    beforeEach(() => {
      onDirectionClick.mockClear();
      wrapper = shallow(<DirectionSwitcher {...props} onDirectionClick={onDirectionClick} />);
      const direction = wrapper.find('li').first();
      direction.simulate('click');
    });

    it('should call onDirectionClick()', () => {
      expect(onDirectionClick.mock.calls.length).toBe(1);
    });

    it('should call onDirectionClick() with correct tag', () => {
      expect(onDirectionClick.mock.calls[0][0]).toBe('tag_1');
    });
  });
});
