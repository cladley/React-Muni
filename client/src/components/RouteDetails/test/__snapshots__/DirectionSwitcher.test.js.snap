// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DirectionSwitcher /> should render correctly 1`] = `
<div
  className="direction-switcher"
>
  <h1>
    1 California
  </h1>
  <ul>
    <li
      onClick={[Function]}
    >
      Inbound to Drumm + Clay
    </li>
    <li
      onClick={[Function]}
    >
      Outbound to Geary + 33rd Avenue
    </li>
  </ul>
</div>
`;
